demo/src/server.ts

```diff
@@ -0,0 +1,67 @@
+import {https} from 'follow-redirects';
+import getStream from 'get-stream';
+import {IncomingMessage} from 'http';
+import Koa from 'koa';
+import koaCompress from 'koa-compress';
+import url from 'url';
+
+import {HTMLRewriter, jsRewrite, PackageJson, parsePackageName} from './html-rewriter';
+import {proxy} from './proxy';
+
+export class RawService {
+  app = new Koa();
+  private port = process.env.PORT || 8080;
+
+  async initalize() {
+    this.app.use(koaCompress());
+
+    this.app.use(this.handleRequest.bind(this));
+
+    return this.app.listen(this.port, () => {
+      console.log(`Listening on port ${this.port}`);
+    });
+  }
+
+  async handleRequest(ctx: Koa.Context, _next: () => {}) {
+    if (ctx.url === '/sw.js' || ctx.url === '/favicon.ico') {
+      return;
+    }
+
+    const proxiedUrl = proxy(ctx.url);
```

Comment:
This is confusing me for some reason. You just use the exact URL from the incoming request and then fetch it as-is? Missing something.